### ISSUE TITLE
S3900 - False Negative on csproj SDK Format

### ISSUE THREAD

**Issue by tnoiretisa** [SonarSource/sonar-dotnet#5236]
### Description

With a csproj on SDK Format, S3900 rules don't throw on SonarLint or on SonarQube (8.9.6 or 9.2.4) after analysis 

### Repro steps

Csproj with SDK Format and sample code not compliant S3900

```
public class MyClass
{
    private MyOtherClass other;

    public void Foo(MyOtherClass other)
    {
        this.other = other; // Compliant: other not being dereferenced
    }

    public void Bar(MyOtherClass other)
    {
        this.other = other.Clone(); // Noncompliant
    }

    protected void FooBar(MyOtherClass other)
    {
        this.other = other.Clone(); // Noncompliant
    }
}
```

### Expected behavior

See issue on SonarLint and on SonarQube

### Actual behavior

No error display

### Related information

* C#/VB.NET Plugins version 8.22.0.31243  & 8.32.0.39516
* Visual Studio version 16.11.8
* MSBuild / dotnet version .net standard 2.0
* SonarScanner for .NET version 5.3.1
* SonarLint 5.3.0.41207

**Comment by costin-zaharia-sonarsource** [comment]
Hi @tnoiretisa, thanks for your feeback. Would it be possible to create a small reproducer for us?

**Comment by costin-zaharia-sonarsource** [comment]
Closing due to inactivity.
